internal/server/SimLoop.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"SOMAS2023/internal/common/objects"
+	"SOMAS2023/internal/common/utils"
+	"SOMAS2023/internal/common/voting"
+	"fmt"
+	"sort"
+
+	"math"
+	"slices"
+
+	"github.com/google/uuid"
+)
+
+func (s *Server) RunSimLoop(iterations int) {
+
+	s.ResetGameState()
+	s.FoundingInstitutions()
+
+	// run this for n iterations
+	for i := 0; i < iterations; i++ {
+		s.RunRoundLoop()
+	}
+
+}
+
+func (s *Server) ResetGameState() {
+	// kick everyone off bikes
+	for id, _ := range s.GetAgentMap() {
+		if bikeId, ok := s.megaBikeRiders[id]; ok {
+			s.megaBikes[bikeId].RemoveAgent(id)
+			delete(s.megaBikeRiders, id)
```

Comment:
This seems incomplete - it doesn't update the bikes' agent list. This is why we need a consistent function to add players to bikes and another to remove players from bikes